### ISSUE TITLE
New errors when playing in MP.

### ISSUE THREAD

**Issue by KITT3000** [Mogli12/GearboxAddon#152]
`Warning (LUA): 'streamWriteInt32': Argument 1 has wrong type. Expected: Int. Actual: Nil 
  ..\..\..\src\base\scripting\lua\LuaScriptSystem.cpp (640): expectedType == Value::VoidType
LUA call stack:
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_GearboxAddon/mogliBase.lua (925) : streamWriteInt32
  dataS/scripts/network/Connection.lua (88) : writeStream
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_GearboxAddon/mogliBase.lua (481) : sendEvent
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_GearboxAddon/gearboxMogli.lua (5985) : mrGbMSetState
  =[C] (4294967295)
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_GearboxAddon/mogliBase.lua (491) : pcall
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_GearboxAddon/mogliBase.lua (538) : mbSetState
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_GearboxAddon/mogliBase.lua (569) : fromMbDocument
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_GearboxAddon/gearboxMogliLoader.lua (90) : readStream
  dataS/scripts/vehicles/Vehicle.lua (1688) : readStream
  dataS/scripts/network/Client.lua (389) : readStream
  dataS/scripts/network/ConnectionManager.lua (28) : func`

`Warning (Script): Unknown entity id 89477 in method 'playSample'.
LUA call stack:
  dataS/scripts/utils/SoundUtil.lua (165) : playSample
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_GearboxAddon/gearboxMogli.lua (3083)
  =[C] (4294967295)
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_GearboxAddon/gearboxMogliLoader.lua (45) : pcall
  dataS/scripts/vehicles/Vehicle.lua (2598) : update
  dataS/scripts/network/Client.lua (64) : update
  dataS/scripts/BaseMission.lua (1511) : update
  dataS/scripts/FSBaseMission.lua (2171) : update
  dataS/scripts/missions/mission00.lua (333) : update
  dataS/scripts/main.lua (1876) : update
Warning (Script): Unknown entity id 89477 in method 'stopSample'.
LUA call stack:
  dataS/scripts/utils/SoundUtil.lua (174) : stopSample
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_GearboxAddon/gearboxMogli.lua (3085)
  =[C] (4294967295)
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_GearboxAddon/gearboxMogliLoader.lua (45) : pcall
  dataS/scripts/vehicles/Vehicle.lua (2598) : update
  dataS/scripts/network/Client.lua (64) : update
  dataS/scripts/BaseMission.lua (1511) : update
  dataS/scripts/FSBaseMission.lua (2171) : update
  dataS/scripts/missions/mission00.lua (333) : update
  dataS/scripts/main.lua (1876) : update`

**Comment by Mogli12** [comment]
I do not see errors I only see warnings!

**Comment by KITT3000** [comment]
Error, warning whatever. There is an entry in the LOG, so something does not work properly, so it has an error.

**Comment by Mogli12** [comment]
What does not work properly? You just posted parts of a log.txt. I do not even know if this is on the server or client and which version you are using. So what do you expect from me?

**Comment by KITT3000** [comment]
Full LOG (client): https://pastebin.com/yrx2cZ59
Warnings appear when the driving mode changes (forwards / backwards). Probably the problem is the backing sound. When the host used the original Manitu wheelloader and changed the mode, these warnings also appeared to him, if only I (client) changed the error mode were only with me.

**Comment by Mogli12** [comment]
I am sure that there is one Mod, which disturbs the complete MP synchronization. The 1st relevant warning is:
```
...
data/vehicles/trailers/newHolland/newHollandRollBelt150.i3d (20.92) ms
Warning (LUA): 'getXMLFloat': Argument 1 has wrong type. Expected: Int. Actual: Nil
  ..\..\..\src\base\scripting\lua\LuaScriptSystem.cpp (640): expectedType == Value::VoidType
LUA call stack:
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_BaleCounter/BaleCounter.lua (22) : getXMLFloat
  dataS/scripts/vehicles/Vehicle.lua (627)
  dataS/scripts/vehicles/Vehicle.lua (1646) : load
  dataS/scripts/network/Client.lua (389) : readStream
  dataS/scripts/network/ConnectionManager.lua (28) : func
Warning (LUA): 'getXMLFloat': Argument 1 has wrong type. Expected: Int. Actual: Nil
  ..\..\..\src\base\scripting\lua\LuaScriptSystem.cpp (640): expectedType == Value::VoidType
LUA call stack:
  =E:/Gry/CDP Games/modyFS17/MP_UserB4D/FS17_BaleCounter/BaleCounter.lua (23) : getXMLFloat
  dataS/scripts/vehicles/Vehicle.lua (627)
  dataS/scripts/vehicles/Vehicle.lua (1646) : load
  dataS/scripts/network/Client.lua (389) : readStream
  dataS/scripts/network/ConnectionManager.lua (28) : func
data/objects/roundbales/roundbaleHay_w112_d130.i3d (10.28) ms
...
```

**Comment by KITT3000** [comment]
Problem with FS17_BaleCounter fixed. 
Full LOG (host): https://pastebin.com/86JBjzeC
Full LOG (client): https://pastebin.com/Tk5JhUxv

**Comment by Mogli12** [comment]
There are still some error messages before the first gearbox error message. But I tried to make the MP coding more robust with version 2.1.2.5. Can you try it?

**Comment by KITT3000** [comment]
Still warnings in LOG, but now only i (client) have warnings when change direction.
Full LOG (host): https://pastebin.com/WtjUWxLS
Full LOG (client): https://pastebin.com/m322Mnxq

**Comment by Mogli12** [comment]
Version 2.1.2.6 should give some more detailed error messages.

**Comment by KITT3000** [comment]
I noticed that the host has a sound when go backwards vehicles, and when I get into the same vehicles I do not have that sound or hear the sound as the host go backwards vehicle.
Full LOG (host): https://pastebin.com/kZCUvZ71
Full LOG (client): https://pastebin.com/Mgz2kvtk

**Comment by Mogli12** [comment]
Can you try it without CutterAttacherHeight? I never finished this mod in FS17. The movement of the cutter makes the combine move up and down.

**Comment by Mogli12** [comment]
The "official" beta version of CutterAttcher ist this one:
https://github.com/Mogli12/CutterAttacherHeight

**Comment by KITT3000** [comment]
With CutterAttcher mod:
Full LOG (host): https://pastebin.com/dCUJkBj6
Full LOG (client): https://pastebin.com/f49LGyNR

Without CutterAttcher mod:
Full LOG (host): https://pastebin.com/1FQ75R6m
Full LOG (client): https://pastebin.com/eK81hZYh

This error (packetReceived)  shown only when client join to the server. In addition, we checked that when the client is moving the vehicle back, even the host does not hear the backing sound (if the host is driving, he hears the sound).

**Comment by Mogli12** [comment]
I do not think that it makes sense to look into this as long as there is this error on the server:
```
Error: Running LUA method 'packetReceived'.
dataS/scripts/vehicles/specializations/events/AnimatedVehicleStartEvent.lua(44) : attempt to index field 'object' (a nil value)
```

**Comment by KITT3000** [comment]
In that case, we will check the evening or the rest of the mods will be ok.

**Comment by KITT3000** [comment]
Only map-mod and Gearebox is active:
Full LOG (host): https://pastebin.com/mxpZ4JD6
Full LOG (client): https://pastebin.com/S7Yc0Ujw

**Comment by Mogli12** [comment]
Hi,
Thanks for the log without all the other mods. The problem should be fixed with 2.1.2.9.
BR
